cabin-sorting/hackerSortingAlgo.ts

```diff
@@ -48,34 +54,45 @@ function matchAnswers () {
     // find backup cabin for hacker (in case the first choice fills up)
     const counterCopy = cabinScore.slice()
     counterCopy[maxIndex] = -1
-    hacker.secondAssignedCabin = cabinOptions[counterCopy.indexOf(Math.max(...counterCopy))]
+    hacker.secondAssignedCabin =
+      cabinOptions[counterCopy.indexOf(Math.max(...counterCopy))]
 
     // show their score for each cabin (for testing purposes)
     console.log(`${hacker.email}'s cabin counter: ${cabinScore}`)
   })
 }
-
+// test
 // Increment the given hacker's given cabinScore each time their answer
 // matches the Cabin's answer
 function hydrateCabinScore (hacker: any, cabinScore: number[]) {
-  answerList.forEach((cabin: any, cabinIndex: number) => {
-    for (let questionIndex = 0; questionIndex < QUESTIONS_SIZE; questionIndex++) {
+  let flag = false
+  for (let questionIndex = 0; questionIndex < QUESTIONS_SIZE; questionIndex++) {
+    flag = false
+    answerList.forEach((cabin: any, cabinIndex: number) => {
       if (
-        cabin['question' + questionIndex.toString()] ===
-        hacker['question' + questionIndex.toString()]
+        cabin['question' + questionIndex].toLowerCase().trim() ===
+        hacker['question' + questionIndex].toLowerCase().trim()
       ) {
         cabinScore[cabinIndex]++
+        flag = true
       }
+    })
+    if (!flag) {
```

Comment:
what is the purpose of this if statement? is flag used for logic anywhere else?

Reply:
The purpose of this if statement is to check if the cabin's answer key is the same as the hacker's answer for this particular question. Flag is not used anywhere else - it's for testing purposes that we will remove